pinot-spi/src/main/java/org/apache/pinot/spi/metrics/PinotMetricProcessor.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.metrics;
+
+
+public interface PinotMetricProcessor<T> {
```

Comment:
Can you add some comments in front of this class stating how it is supposed to be used? And/or add some comments in front of each method stating what `process()` means. Thanks

Reply:
Java doc descriptions are added to all the interfaces.

Reply:
+1 to @mcvsubbu's comments. Can you add comments on how one can use the interface and plug in different metrics libs for example DropWizard?

Reply:
I checked the usage of `MetricProcessor` in Pinot existing code. It's used in Aggregate metrics like `AggregatedCounter`, `AggregatedMeter`,  `AggregatedHistogram`. But since these metrics are legacy code, we actually don't need this interface any more. I've removed this interface in the next push.